components/script/dom/webglrenderingcontext.rs

```diff
@@ -1281,15 +1290,26 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
             return self.webgl_error(InvalidOperation);
         }
 
-        if count <= 0 {
-            return self.webgl_error(InvalidOperation);
+        if count < 0 {
+            return self.webgl_error(InvalidValue);
         }
 
         if offset < 0 {
             return self.webgl_error(InvalidValue);
         }
 
-        if self.current_program.get().is_none() || self.bound_buffer_element_array.get().is_none() {
+        if self.current_program.get().is_none() {
```

Comment:
I'll pass the tests again without checking current_program & bound_buffer_element_array. Just think that some webgl conformance tests expect a error in that cases, but I want to to confirm.

When there is an overflow we must generate the error for sure. The are a lot of tests in the webgl conformance about overflows.